### PR TITLE
pybind/rados: add support open_ioctx2 API

### PR DESCRIPTION
**Pull request by Songweibin**
Can use `client.open_ioct2(pool_id)` to open an IO.

Signed-off-by: songweibin <song.weibin@zte.com.cn>

### CODE REVIEW

**Comment by scienceluo** [comment]
jenkins retest this please.

**Comment by Songweibin** [comment]
@jdurgin  Would you please help to take a look? Thanks in advance.

File: doc/rbd/api/librbdpy.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ context::
     cluster.connect()
     ioctx = cluster.open_ioctx('mypool')
 
+Also, you can use `open_ioctx2(mypool_id)` method to open an IO.
```
Comment (tchaikov) [inline doc/rbd/api/librbdpy.rst]: no need to put this line in the example. it is not supposed to be exhaustive.

**Comment by tchaikov** [comment]
would be better to accompany this change with a test in `src/test/pybind/test_rados.py`.

**Comment by Songweibin** [comment]
@tchaikov  Done, added a full test just like [`class TestIoctx(object)`](https://github.com/ceph/ceph/blob/master/src/test/pybind/test_rados.py#L277), Thank you.

File: src/test/pybind/test_rados.py [hunk 2]
```diff
@@ -891,6 +891,627 @@ def test_applications(self):
         self.ioctx.application_metadata_remove("app1", "key1")
         eq([("key2", "val2")], self.ioctx.application_metadata_list("app1"))
 
+
+class TestIoctx2(object):
```
Comment (tchaikov) [inline src/test/pybind/test_rados.py]: i don't think we need to duplicate the whole `TestIoctx` just for testing `rados.open_ioctx2`.

Comment (Songweibin) [reply src/test/pybind/test_rados.py]: yea! i do not want to take a whole test at the beginning, but i have no idea which ones should be testing, so...your suggestions is appreciate. thanks.

**Comment by Songweibin** [comment]
@tchaikov  only take a simple test, would you please help to have a look again? thanks.

**Pull request merged**